### DISCUSSION TITLE
Problems updating firmware

### DISCUSSION THREAD

**Discussion by simplect** [Jyers/Marlin#610] [Q&A] [answered]
Hi!

I am currently on 1.2.1. And after formatting the sdcard (I do not know if that is the cause of the problem), I do not seem to be able to update to the latest version. The boot is very quick (seems like it doesnt update) and in the info it still says 1.2.1. I formatted the SDcard to FAT32. 

Any suggestions what I can try?

Thanks!

**Comment by Tuncay-Ayhan** [comment, marked as answer]
Rename the firmware.bin file to something else. You can't reuse the same name
twice when updating.

Edit: 

Spelling

**Reply by simplect** [reply]
Thank you! Will try that.

**Comment by wozzo** [comment]
@Jyers I came here to suggest adding the release version to the bin file names to avoid this happening.
e.g. E3V2-BLTouch-3x3-HS-v4.2.2-v1.3.2.bin instead of E3V2-BLTouch-3x3-HS-v4.2.2.bin or something similar.
What do you think?

**Reply by Jyers** [reply]
It just depends whether I can automate it. Its been on my todo list to check out, but it keeps getting pushed down 😅

**Reply by Tuncay-Ayhan** [reply]
> @Jyers I came here to suggest adding the release version to the bin file names to avoid this happening.
> e.g. E3V2-BLTouch-3x3-HS-v4.2.2-v1.3.2.bin instead of E3V2-BLTouch-3x3-HS-v4.2.2.bin or something similar.
> What do you think?

I've created a pull-request and it has been merged. So from the next release on, you will have version tags in the filenames.

**Reply by wozzo** [reply]
Star

**Comment by Tuncay-Ayhan** [comment]
Someone else asked that also

https://github.com/Jyers/Marlin/issues/541#issuecomment-822979153

**Reply by wozzo** [reply]
Thanks.

**Comment by smooreace** [comment]
Not likely going to happen. And if you want files named differently then
name them differently yourself. Everyone has different filing processes!
Very few *should* be depending on the file name to define its version!

**Reply by wozzo** [reply]
Sure. But in this case it's a requirement of the hardware that the files be named differently. If the files are all the same names then there'll be lots of people who will struggle to upgrade.
I'm happy to rename myself but just trying to help out others who aren't aware of the different name requirement

**Comment by smooreace** [comment]
I don't disagree. But one of the first keys that every guide and video
mentions is that firmware files have to have a different name than that
last installed. It's also one of the first troubleshooting tips in most
guides. So if someone has trouble it's simple through laziness. There is no
simple process to automatically add a version number to all of the files...
unless a variable can be defined and injected into the automated compiling
processes on GitHub. And as far as I know with my limited background, there
isn't It would be absurd to expect Jyers to manually edit all those files
just to save a few lazy people the embarrassment of having to ask a dumb
question.

**Reply by simplect** [reply]
Usually I have enough previous experience in embedded engineering to quickly troubleshoot most problems I come across. For this problem I tried googling a few times (for example "can't flash firmware ender 3 v2"), and most problems that pop-up are related to bricks or black screens. Previously I had no problems flashing this firmware, but when Jyers changed to the build-in Github release system it lacked the version numbers.

It's always better to help each other to prevent "dumb" and "lazy" mistakes, instead of hiding behind product specific knowledge and calling outsiders lazy/dumb. 

If adding versions names requires manual renaming of all the files, then it's obviously not feasible. It's an option to add a note somewhere, but a great community, like @Tuncay-Ayhan  here, helps solve the problem quickly.

**Comment by smooreace** [comment]
Dumb and lazy are global! It doesn't matter if it's reading instructions on
a glue bottle or building a turbine engine or Chinese rocket scientists!
It's hardly a product-specific knowledge/issue! It's, in general, the
global spoon-feeding of literally everything that has created a population
dependent on that spoon-feeding. Everyone fails sometimes. What sets adults
apart is their ability to learn and grow through their own actions vs those
that actively refuse to grow, or try.

**Comment by Tuncay-Ayhan** [comment]
Well... There were so many people creating issues because they couldn't update without changing the filename...

So I've added a pull-request adding versions to the pre-compiled firmware https://github.com/Jyers/Marlin/pull/752

If all goes well, it should be available from the next release on.

**Comment by smooreace** [comment]
Thats the kind of response people can benefit from!

Unfortunately, lazy people will still fail to follow instructions and ask
the same *NEXT *question. I wish I had learned this crap decades ago!
